PR: https://github.com/odoo/odoo/pull/

From: a3dda5b350a505786c8408326b87c55db6fa8085
From: Prakash Prajapati
Date: 2018-02-22 14:19:51

Structural Changes: 5
Total Changes: 22

[IMP] base, base_iban, l10n_ch: few improvements in res.partner.bank

- Change the type in 'acc_type' field char into selection field and change tooltip
- added new field 'acc_holder_name' that can be used to set the account holder name in case it is different than the partner name

Was task 39334.
Was PR #20794

================================= pseudo patch: =================================

--- a/odoo/addons/base/models/res_bank.py
+++ b/odoo/addons/base/models/res_bank.py
@@ -59,10 +59,11 @@ class ResPartnerBank(models.Model):
     _description = 'Bank Accounts'
     _order = 'sequence'
 
-    acc_type = fields.Char(compute='_compute_acc_type', help='Bank account type, inferred from account number')
+    acc_type = fields.Selection([('bank', 'Normal')], compute='_compute_acc_type', string='Type', help='Bank account type: Normal or IBAN. Inferred from the bank account number.')
     acc_number = fields.Char('Account Number', required=True)
     sanitized_acc_number = fields.Char(compute='_compute_sanitized_acc_number', string='Sanitized Account Number', readonly=True, store=True)
-    partner_id = fields.Many2one('res.partner', 'Account Holder', ondelete='cascade', index=True, domain=['|', ('is_company', '=', True), ('parent_id', '=', False)], default=lambda self: self.env.user.company_id.partner_id)
+    acc_holder_name = fields.Char(string='Account Holder Name', help="Account holder name, in case it is different than the name of the Account Holder")
+    partner_id = fields.Many2one('res.partner', 'Account Holder', ondelete='cascade', index=True, domain=['|', ('is_company', '=', True), ('parent_id', '=', False)])
     bank_id = fields.Many2one('res.bank', string='Bank')
     bank_name = fields.Char(related='bank_id.name')
     bank_bic = fields.Char(related='bank_id.bic')

--- a/odoo/addons/base/views/res_bank_views.xml
+++ b/odoo/addons/base/views/res_bank_views.xml
@@ -62,11 +62,18 @@
             <field name="model">res.partner.bank</field>
             <field name="arch" type="xml">
                 <form string="Bank account">
-                    <group col="4">
-                        <field name="acc_number"/>
-                        <field name="bank_id"/>
-                        <field name="currency_id" groups="base.group_multi_currency" options="{'no_create': True}"/>
-                        <field name="company_id" groups="base.group_multi_company" options="{'no_create': True}"/>
+                    <group>
+                        <group>
+                            <field name="acc_number"/>
+                            <field name="acc_type"/>
+                            <field name="company_id" groups="base.group_multi_company" options="{'no_create': True}"/>
+                            <field name="partner_id"/>
+                        </group>
+                        <group>
+                            <field name="bank_id"/>
+                            <field name="currency_id" groups="base.group_multi_currency" options="{'no_create': True}"/>
+                            <field name="acc_holder_name" groups="base.group_no_one"/>
+                        </group>
                     </group>
                 </form>
             </field>
